include/revng/Pipeline/Rank.h

```diff
@@ -16,17 +16,28 @@ namespace pipeline {
 /// The rank tree is a tree used by targets to find out how many names
 /// are required to name a target
 class Rank : public DynamicHierarchy<Rank> {
+private:
+  std::string ModelPath;
+
 public:
   Rank(llvm::StringRef Name) : DynamicHierarchy(Name) {}
+  Rank(llvm::StringRef Name, std::string_view ModelPath) :
+    DynamicHierarchy(Name), ModelPath(ModelPath) {}
   Rank(llvm::StringRef Name, Rank &Parent) : DynamicHierarchy(Name, Parent) {}
+  Rank(llvm::StringRef Name, Rank &Parent, std::string_view ModelPath) :
+    DynamicHierarchy(Name, Parent), ModelPath(ModelPath) {}
+
+  std::string_view modelPath() const { return ModelPath; }
 };
 
 // Root rank specialization
-template<ConstexprString String>
+template<ConstexprString String, ConstexprString ModelPathSeparator_>
 class RootRank : public Rank {
 public:
   static constexpr bool RankTag = true;
   static constexpr std::string_view RankName = String;
+  static constexpr std::string_view ModelPathSeparator = ModelPathSeparator_;
```

Comment:
I really dislike making something generic like this aware of the model specifics, this lets the model separator be encoded in `revng/Pipes/Ranks.h` which I feel is a lot more reasonable.

Reply:
This is not model-specific, it's `TupleTree`-specific. We hardcoded `/` as a `TupleTreePath` separator everywhere.

Also keep in mind that revng-pipeline is in maintenance mode while waiting for revng-pypeline (note the py).
I'm really not keen of adding any new complexity.

Reply:
> This is not model-specific, it's `TupleTree`-specific.

ah, wait, right. My bad then.

> also keep in mind that revng-pipeline is in maintenance mode while waiting for revng-pypeline (note the py).

Yes, but I'm pretty sure some semblance of the location infrastructure will still stay (we will always have to both parse and make locations from the cpp-side), and this (the mapping to the model paths) is kind of relevant to it, even if we managed to avoid using it for now.

Reply:
Ping

Reply:
This is done, but I'm not allowed to close a thread, so please do it for me.